tests/utils.test.ts

```diff
@@ -17,4 +17,82 @@ describe('calculateLogMessageSize()', () => {
     expect(calculateLogMessageSize(STRING_SIZE_7546_BYTES)).toBe(7546);
     expect(calculateLogMessageSize({ details: STRING_SIZE_7546_BYTES })).toBe(7558);
   });
+});
+
+describe('deepClone()', () => {
+  it('should clone a passed in item deeply; primitive value', () => {
+    const testItem = 3;
+    const clonedTest = deepClone(testItem);
+    expect(clonedTest).toBe(3);
+    expect(typeof clonedTest).toBe('number');
+  });
+  it('should clone a passsed in item deeply; basic array', () => {
+    const testItem = [1, 2, 3];
+    const clonedTest = deepClone(testItem);
+    expect(clonedTest).toStrictEqual(testItem);
+    expect(typeof clonedTest).toBe('object');
+    expect(Array.isArray(clonedTest)).toBeTruthy();
+  });
+  it('should clone a passed in item deeply; complex array', () => {
+    const testItem = [1, "hello", [1, 2, 3], function() { console.log('hello')}, {test1: 1, test2: 2}];
+    const clonedTest = deepClone(testItem);
+    expect(clonedTest).toStrictEqual(testItem);
+    expect(typeof clonedTest).toBe('object');
+    expect(Array.isArray(clonedTest)).toBeTruthy();
+    expect(Array.isArray(clonedTest[2])).toBeTruthy();
+    expect(typeof clonedTest[3]).toBe('function');
+    expect(typeof clonedTest[4]).toBe('object');
+    expect(Object.keys(clonedTest[4]).length).toBe(2);
+  });
+  it('should clone a passed in item deeply; simple object', () => {
+    const testItem = {test1: 1, test2: 2};
+    const clonedTest = deepClone(testItem);
+    expect(clonedTest).toStrictEqual(testItem);
+    expect(typeof clonedTest).toBe('object');
+    expect(Object.keys(clonedTest).length).toBe(2);
+    expect(clonedTest.test1).toBe(1);
+    expect(clonedTest.test2).toBe(2);
+  });
+  it('should clone a passed in item deeply; complex object', () => {
+    const testItem = {
+      test1: 1,
+      testString: 'hello',
+      testObj: {
+        test2: 2,
+        testString2: 'hello',
+        testObj2: {
+          test3: 3,
+          testString3: 'hello'
+        }
+      },
+      testFunc: function() {
+        console.log('hello');
+      },
+      testArray: [1, 2, 3, 4]
+    }
+    const clonedTest = deepClone(testItem);
+    expect(clonedTest).toStrictEqual(testItem);
+    expect(typeof clonedTest).toBe('object');
+    expect(Object.keys(clonedTest).length).toBe(5);
+    expect(typeof clonedTest.testObj).toBe('object');
+    expect(Object.keys(clonedTest.testObj).length).toBe(3);
+    expect(typeof clonedTest.testObj.testObj2).toBe('object');
+    expect(Object.keys(clonedTest.testObj.testObj2).length).toBe(2);
+    expect(typeof clonedTest.testFunc).toBe('function');
+    expect(typeof clonedTest.testArray).toBe('object');
+    expect(Array.isArray(clonedTest.testArray)).toBeTruthy();
```

Comment:
Not that you need to change this, but I think these expects could be simplified to something like this: 
``` ts
const testItem = { ... };
const clonedTest = deepClone(testItem);

expect(clonedTest).toStrictEqual(testItem); // could also use `toEqual` but `toStrictEqual` is fine too (and maybe more thorough)
expect(clonedTest).not.toBe(testItem); // make sure we don't have the same object reference
expect(clonedTest.testObj).not.toBe(testItem.testObj); // check to make sure object ref is not the same
expect(clonedTest.testArray).not.toBe(testItem.testArray); // check to make sure object ref is not the same
```